### ISSUE TITLE
metrics prefix inconsistent, should be "minio_" for minio-specific metrics

### ISSUE THREAD

**Issue by briend** [minio/minio#11225]
**Is your feature request related to a problem? Please describe.**

Some of the minio metrics are namespaced with `minio_`.  Metrics like `minio_disks_offline` make a lot of sense. However there are others that really ought to have the `minio_` prefix, such as `self_heal_objects_healed`, `bucket_objects_count`, `internode_rx_bytes_total`, and others.
Having a common prefix for most of the metrics makes it much easier to perform adhoc queries in the prometheus web client, since you can simply type "minio_" and see all the relevant metrics names. Downside to making this change would be breaking current dashboards and prometheus alerting rules, etc.

**Describe the solution you'd like**
All metrics that are specific to minio should be namespaced according to the prometheus recommendations:

https://prometheus.io/docs/practices/naming/#metric-names

>..should have a (single-word) application prefix relevant to the domain the metric belongs to. The prefix is sometimes referred to as namespace by client libraries. For metrics specific to an application, the prefix is usually the application name itself. Sometimes, however, metrics are more generic, like standardized metrics exported by client libraries. Examples:
prometheus_notifications_total (specific to the Prometheus server)
process_cpu_seconds_total (exported by many client libraries)
http_request_duration_seconds (for all HTTP requests)

**Describe alternatives you've considered**
I'm on the fence about the `s3_rx_bytes_total` and other `s3_` metrics.  Should those be `minio_s3`?  I'm not quite sure, but unless the `s3_` prefix is already commonly used by other s3 applications, I don't see the benefit in making a generic metric for these.

**Additional context**

current list of metrics from our server running minio `2020-12-29T23:29:29Z`:

```
# HELP bucket_objects_count Total number of objects in a bucket
# TYPE bucket_objects_count gauge
bucket_objects_count{bucket="test"} 18091
# HELP bucket_objects_histogram Total number of objects of different sizes in a bucket
# TYPE bucket_objects_histogram gauge
bucket_objects_histogram{bucket="test",object_size="BETWEEN_1024_B_AND_1_MB"} 739
bucket_objects_histogram{bucket="test",object_size="BETWEEN_10_MB_AND_64_MB"} 228
bucket_objects_histogram{bucket="test",object_size="BETWEEN_128_MB_AND_512_MB"} 55
bucket_objects_histogram{bucket="test",object_size="BETWEEN_1_MB_AND_10_MB"} 76
bucket_objects_histogram{bucket="test",object_size="BETWEEN_64_MB_AND_128_MB"} 63
bucket_objects_histogram{bucket="test",object_size="GREATER_THAN_512_MB"} 0
bucket_objects_histogram{bucket="test",object_size="LESS_THAN_1024_B"} 16930
# HELP bucket_usage_size Total bucket size
# TYPE bucket_usage_size gauge
bucket_usage_size{bucket="test"} 2.4022947612e+10
# HELP disk_storage_available Total available space left on the disk
# TYPE disk_storage_available gauge
disk_storage_available{disk="/export"} 7.21406476288e+11
# HELP disk_storage_total Total space on the disk
# TYPE disk_storage_total gauge
disk_storage_total{disk="/export"} 7.34974173184e+11
# HELP disk_storage_used Total disk storage used on the disk
# TYPE disk_storage_used gauge
disk_storage_used{disk="/export"} 1.3567696896e+10
# HELP go_gc_duration_seconds A summary of the GC invocation durations.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 7.1331e-05
go_gc_duration_seconds{quantile="0.25"} 0.0001226
go_gc_duration_seconds{quantile="0.5"} 0.000154087
go_gc_duration_seconds{quantile="0.75"} 0.000196795
go_gc_duration_seconds{quantile="1"} 0.00790421
go_gc_duration_seconds_sum 2.463813069
go_gc_duration_seconds_count 11881
# HELP go_goroutines Number of goroutines that currently exist.
# TYPE go_goroutines gauge
go_goroutines 167
# HELP go_info Information about the Go environment.
# TYPE go_info gauge
go_info{version="go1.15.5"} 1
# HELP go_memstats_alloc_bytes Number of bytes allocated and still in use.
# TYPE go_memstats_alloc_bytes gauge
go_memstats_alloc_bytes 3.00359136e+08
# HELP go_memstats_alloc_bytes_total Total number of bytes allocated, even if freed.
# TYPE go_memstats_alloc_bytes_total counter
go_memstats_alloc_bytes_total 1.110974997264e+12
# HELP go_memstats_buck_hash_sys_bytes Number of bytes used by the profiling bucket hash table.
# TYPE go_memstats_buck_hash_sys_bytes gauge
go_memstats_buck_hash_sys_bytes 1.9617251e+07
# HELP go_memstats_frees_total Total number of frees.
# TYPE go_memstats_frees_total counter
go_memstats_frees_total 6.951404341e+09
# HELP go_memstats_gc_cpu_fraction The fraction of this program's available CPU time used by the GC since the program started.
# TYPE go_memstats_gc_cpu_fraction gauge
go_memstats_gc_cpu_fraction 1.2816095019495712e-05
# HELP go_memstats_gc_sys_bytes Number of bytes used for garbage collection system metadata.
# TYPE go_memstats_gc_sys_bytes gauge
go_memstats_gc_sys_bytes 2.7111696e+07
# HELP go_memstats_heap_alloc_bytes Number of heap bytes allocated and still in use.
# TYPE go_memstats_heap_alloc_bytes gauge
go_memstats_heap_alloc_bytes 3.00359136e+08
# HELP go_memstats_heap_idle_bytes Number of heap bytes waiting to be used.
# TYPE go_memstats_heap_idle_bytes gauge
go_memstats_heap_idle_bytes 3.52477184e+08
# HELP go_memstats_heap_inuse_bytes Number of heap bytes that are in use.
# TYPE go_memstats_heap_inuse_bytes gauge
go_memstats_heap_inuse_bytes 3.13204736e+08
# HELP go_memstats_heap_objects Number of allocated objects.
# TYPE go_memstats_heap_objects gauge
go_memstats_heap_objects 123705
# HELP go_memstats_heap_released_bytes Number of heap bytes released to OS.
# TYPE go_memstats_heap_released_bytes gauge
go_memstats_heap_released_bytes 1.98623232e+08
# HELP go_memstats_heap_sys_bytes Number of heap bytes obtained from system.
# TYPE go_memstats_heap_sys_bytes gauge
go_memstats_heap_sys_bytes 6.6568192e+08
# HELP go_memstats_last_gc_time_seconds Number of seconds since 1970 of last garbage collection.
# TYPE go_memstats_last_gc_time_seconds gauge
go_memstats_last_gc_time_seconds 1.6098675924871442e+09
# HELP go_memstats_lookups_total Total number of pointer lookups.
# TYPE go_memstats_lookups_total counter
go_memstats_lookups_total 0
# HELP go_memstats_mallocs_total Total number of mallocs.
# TYPE go_memstats_mallocs_total counter
go_memstats_mallocs_total 6.951528046e+09
# HELP go_memstats_mcache_inuse_bytes Number of bytes in use by mcache structures.
# TYPE go_memstats_mcache_inuse_bytes gauge
go_memstats_mcache_inuse_bytes 27776
# HELP go_memstats_mcache_sys_bytes Number of bytes used for mcache structures obtained from system.
# TYPE go_memstats_mcache_sys_bytes gauge
go_memstats_mcache_sys_bytes 32768
# HELP go_memstats_mspan_inuse_bytes Number of bytes in use by mspan structures.
# TYPE go_memstats_mspan_inuse_bytes gauge
go_memstats_mspan_inuse_bytes 710600
# HELP go_memstats_mspan_sys_bytes Number of bytes used for mspan structures obtained from system.
# TYPE go_memstats_mspan_sys_bytes gauge
go_memstats_mspan_sys_bytes 3.227648e+06
# HELP go_memstats_next_gc_bytes Number of heap bytes when next garbage collection will take place.
# TYPE go_memstats_next_gc_bytes gauge
go_memstats_next_gc_bytes 4.05618608e+08
# HELP go_memstats_other_sys_bytes Number of bytes used for other system allocations.
# TYPE go_memstats_other_sys_bytes gauge
go_memstats_other_sys_bytes 3.623253e+06
# HELP go_memstats_stack_inuse_bytes Number of bytes in use by the stack allocator.
# TYPE go_memstats_stack_inuse_bytes gauge
go_memstats_stack_inuse_bytes 5.40672e+06
# HELP go_memstats_stack_sys_bytes Number of bytes obtained from system for stack allocator.
# TYPE go_memstats_stack_sys_bytes gauge
go_memstats_stack_sys_bytes 5.40672e+06
# HELP go_memstats_sys_bytes Number of bytes obtained from system.
# TYPE go_memstats_sys_bytes gauge
go_memstats_sys_bytes 7.24701256e+08
# HELP go_threads Number of OS threads created.
# TYPE go_threads gauge
go_threads 55
# HELP internode_rx_bytes_total Total number of internode bytes received by current MinIO server instance
# TYPE internode_rx_bytes_total counter
internode_rx_bytes_total 1.5044723091e+10
# HELP internode_tx_bytes_total Total number of bytes sent to the other peer nodes by current MinIO server instance
# TYPE internode_tx_bytes_total counter
internode_tx_bytes_total 2.2894529286e+10
# HELP minio_disks_offline Total number of offline disks in current MinIO server instance
# TYPE minio_disks_offline gauge
minio_disks_offline 0
# HELP minio_disks_total Total number of disks for current MinIO server instance
# TYPE minio_disks_total gauge
minio_disks_total 1
# HELP minio_version_info Version of current MinIO server instance
# TYPE minio_version_info gauge
minio_version_info{commit="951b6b203b85995dc076c0e3efe934febd6cdcca",version="2020-12-03T05:49:24Z"} 1
# HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
# TYPE process_cpu_seconds_total counter
process_cpu_seconds_total 26000.01
# HELP process_max_fds Maximum number of open file descriptors.
# TYPE process_max_fds gauge
process_max_fds 1.048576e+06
# HELP process_open_fds Number of open file descriptors.
# TYPE process_open_fds gauge
process_open_fds 70
# HELP process_resident_memory_bytes Resident memory size in bytes.
# TYPE process_resident_memory_bytes gauge
process_resident_memory_bytes 6.5794048e+08
# HELP process_start_time_seconds Start time of the process since unix epoch in seconds.
# TYPE process_start_time_seconds gauge
process_start_time_seconds 1.60867836285e+09
# HELP process_virtual_memory_bytes Virtual memory size in bytes.
# TYPE process_virtual_memory_bytes gauge
process_virtual_memory_bytes 1.420926976e+09
# HELP process_virtual_memory_max_bytes Maximum amount of virtual memory available in bytes.
# TYPE process_virtual_memory_max_bytes gauge
process_virtual_memory_max_bytes -1
# HELP promhttp_metric_handler_requests_in_flight Current number of scrapes being served.
# TYPE promhttp_metric_handler_requests_in_flight gauge
promhttp_metric_handler_requests_in_flight 1
# HELP promhttp_metric_handler_requests_total Total number of scrapes by HTTP status code.
# TYPE promhttp_metric_handler_requests_total counter
promhttp_metric_handler_requests_total{code="200"} 158301
promhttp_metric_handler_requests_total{code="500"} 0
promhttp_metric_handler_requests_total{code="503"} 0
# HELP s3_errors_total Total number of s3 errors in current MinIO server instance
# TYPE s3_errors_total counter
s3_errors_total{api="getobject"} 108
# HELP s3_requests_current Total number of running s3 requests in current MinIO server instance
# TYPE s3_requests_current counter
s3_requests_current{api="completemutipartupload"} 0
s3_requests_current{api="copyobject"} 0
s3_requests_current{api="copyobjectpart"} 2
s3_requests_current{api="deletemultipleobjects"} 0
s3_requests_current{api="getobject"} 0
s3_requests_current{api="listmultipartuploads"} 0
s3_requests_current{api="listobjectparts"} 0
s3_requests_current{api="listobjectsv2"} 0
s3_requests_current{api="newmultipartupload"} 0
s3_requests_current{api="putobject"} 0
s3_requests_current{api="putobjectpart"} 0
# HELP s3_requests_total Total number of s3 requests in current MinIO server instance
# TYPE s3_requests_total counter
s3_requests_total{api="completemutipartupload"} 294
s3_requests_total{api="copyobject"} 93
s3_requests_total{api="copyobjectpart"} 141
s3_requests_total{api="deletemultipleobjects"} 363
s3_requests_total{api="getobject"} 2510
s3_requests_total{api="listmultipartuploads"} 412
s3_requests_total{api="listobjectparts"} 409
s3_requests_total{api="listobjectsv2"} 33770
s3_requests_total{api="newmultipartupload"} 252
s3_requests_total{api="putobject"} 1606
s3_requests_total{api="putobjectpart"} 547
# HELP s3_rx_bytes_total Total number of s3 bytes received by current MinIO server instance
# TYPE s3_rx_bytes_total counter
s3_rx_bytes_total 4.32960346e+09
# HELP s3_ttfb_seconds Time taken by requests served by current MinIO server instance
# TYPE s3_ttfb_seconds histogram
s3_ttfb_seconds_bucket{api="getobject",le="0.05"} 2299
s3_ttfb_seconds_bucket{api="getobject",le="0.1"} 2304
s3_ttfb_seconds_bucket{api="getobject",le="0.25"} 2318
s3_ttfb_seconds_bucket{api="getobject",le="0.5"} 2383
s3_ttfb_seconds_bucket{api="getobject",le="1"} 2442
s3_ttfb_seconds_bucket{api="getobject",le="2.5"} 2484
s3_ttfb_seconds_bucket{api="getobject",le="5"} 2506
s3_ttfb_seconds_bucket{api="getobject",le="10"} 2510
s3_ttfb_seconds_bucket{api="getobject",le="+Inf"} 2510
s3_ttfb_seconds_sum{api="getobject"} 251.70517151699966
s3_ttfb_seconds_count{api="getobject"} 2510
s3_ttfb_seconds_bucket{api="listmultipartuploads",le="0.05"} 412
s3_ttfb_seconds_bucket{api="listmultipartuploads",le="0.1"} 412
s3_ttfb_seconds_bucket{api="listmultipartuploads",le="0.25"} 412
s3_ttfb_seconds_bucket{api="listmultipartuploads",le="0.5"} 412
s3_ttfb_seconds_bucket{api="listmultipartuploads",le="1"} 412
s3_ttfb_seconds_bucket{api="listmultipartuploads",le="2.5"} 412
s3_ttfb_seconds_bucket{api="listmultipartuploads",le="5"} 412
s3_ttfb_seconds_bucket{api="listmultipartuploads",le="10"} 412
s3_ttfb_seconds_bucket{api="listmultipartuploads",le="+Inf"} 412
s3_ttfb_seconds_sum{api="listmultipartuploads"} 1.521800934
s3_ttfb_seconds_count{api="listmultipartuploads"} 412
s3_ttfb_seconds_bucket{api="listobjectparts",le="0.05"} 409
s3_ttfb_seconds_bucket{api="listobjectparts",le="0.1"} 409
s3_ttfb_seconds_bucket{api="listobjectparts",le="0.25"} 409
s3_ttfb_seconds_bucket{api="listobjectparts",le="0.5"} 409
s3_ttfb_seconds_bucket{api="listobjectparts",le="1"} 409
s3_ttfb_seconds_bucket{api="listobjectparts",le="2.5"} 409
s3_ttfb_seconds_bucket{api="listobjectparts",le="5"} 409
s3_ttfb_seconds_bucket{api="listobjectparts",le="10"} 409
s3_ttfb_seconds_bucket{api="listobjectparts",le="+Inf"} 409
s3_ttfb_seconds_sum{api="listobjectparts"} 3.2197313129999974
s3_ttfb_seconds_count{api="listobjectparts"} 409
s3_ttfb_seconds_bucket{api="listobjectsv2",le="0.05"} 33644
s3_ttfb_seconds_bucket{api="listobjectsv2",le="0.1"} 33667
s3_ttfb_seconds_bucket{api="listobjectsv2",le="0.25"} 33674
s3_ttfb_seconds_bucket{api="listobjectsv2",le="0.5"} 33674
s3_ttfb_seconds_bucket{api="listobjectsv2",le="1"} 33703
s3_ttfb_seconds_bucket{api="listobjectsv2",le="2.5"} 33757
s3_ttfb_seconds_bucket{api="listobjectsv2",le="5"} 33769
s3_ttfb_seconds_bucket{api="listobjectsv2",le="10"} 33770
s3_ttfb_seconds_bucket{api="listobjectsv2",le="+Inf"} 33770
s3_ttfb_seconds_sum{api="listobjectsv2"} 585.3078568260029
s3_ttfb_seconds_count{api="listobjectsv2"} 33770
# HELP s3_tx_bytes_total Total number of s3 bytes sent by current MinIO server instance
# TYPE s3_tx_bytes_total counter
s3_tx_bytes_total 1.1535551232e+10
# HELP self_heal_objects_healed Objects healed in current self healing run
# TYPE self_heal_objects_healed gauge
self_heal_objects_healed{type="object"} 186
# HELP self_heal_objects_scanned Objects scanned in current self healing run
# TYPE self_heal_objects_scanned gauge
self_heal_objects_scanned{type="bucket-metadata"} 186
# HELP self_heal_time_since_last_activity Time elapsed (in nano seconds) since last self healing activity. This is set to -1 until initial self heal activity
# TYPE self_heal_time_since_last_activity gauge
self_heal_time_since_last_activity 1.189108655550499e+15
```

**Comment by harshavardhana** [comment]
They do take the main namespace on purpose this was already agreed and decided as a correct thing to do. We will check internally and let you know if we can namespace them appropriately. 

It might be a breaking change

**Comment by mkhpalm** [comment]
For me it would make the most sense if all minio metrics are prefixed with `minio_` including `minio_s3_*` since there is no consensus / standard metric name + labels among s3 exporter implementations. (minio, ceph, gluster s3, etc) Doing so would align with the prometheus recommendations but retain freedom for minio devs working on the exported metrics in the future.

**Comment by zapto2600** [comment]
+1 to this, the name space needs to be Minio_ We import the metrics into our own prometheus instance and its hard to wade through to find the right MinIO instances.

**Comment by harshavardhana** [comment]
Fixed by #11141
